src/borg/testsuite/nonces.py

```diff
@@ -79,146 +59,137 @@ def set_cache_nonce(self, nonce):
     def test_empty_cache_and_old_server(self, monkeypatch):
         monkeypatch.setattr(nonces, 'NONCE_SPACE_RESERVATION', 0x20)
 
-        cipher = self.MockCipher(0x2000)
         self.repository = self.MockOldRepository()
-        manager = NonceManager(self.repository, cipher, 0x2000)
-        manager.ensure_reservation(19)
-        cipher.expect_iv_and_advance(0x2000, 0x2013)
+        manager = NonceManager(self.repository, 0x2000)
+        next_nonce = manager.ensure_reservation(0x2000, 19)
+        assert next_nonce == 0x2000
 
         assert self.cache_nonce() == "0000000000002033"
 
     def test_empty_cache(self, monkeypatch):
         monkeypatch.setattr(nonces, 'NONCE_SPACE_RESERVATION', 0x20)
 
-        cipher = self.MockCipher(0x2000)
         self.repository = self.MockRepository()
         self.repository.next_free = 0x2000
-        manager = NonceManager(self.repository, cipher, 0x2000)
-        manager.ensure_reservation(19)
-        cipher.expect_iv_and_advance(0x2000, 0x2013)
+        manager = NonceManager(self.repository, 0x2000)
+        next_nonce = manager.ensure_reservation(0x2000, 19)
+        assert next_nonce == 0x2000
 
         assert self.cache_nonce() == "0000000000002033"
 
     def test_empty_nonce(self, monkeypatch):
         monkeypatch.setattr(nonces, 'NONCE_SPACE_RESERVATION', 0x20)
 
-        cipher = self.MockCipher(0x2000)
         self.repository = self.MockRepository()
         self.repository.next_free = None
-        manager = NonceManager(self.repository, cipher, 0x2000)
-        manager.ensure_reservation(19)
-        cipher.expect_iv_and_advance(0x2000, 0x2000 + 19)
+        manager = NonceManager(self.repository, 0x2000)
+        next_nonce = manager.ensure_reservation(0x2000, 19)
+        assert next_nonce == 0x2000
 
         assert self.cache_nonce() == "0000000000002033"
         assert self.repository.next_free == 0x2033
 
         # enough space in reservation
-        manager.ensure_reservation(13)
-        cipher.expect_iv_and_advance(0x2013, 0x2000 + 19 + 13)
+        next_nonce = manager.ensure_reservation(0x2013, 13)
```

Comment:
One way to use it would be to simulate the ciphers iv tracking. i.e.
```
next_nonce = manager.ensure_reservation(nonce, 13)
assert next_nonce == nonce
nonce += 13
```
For where the nonce isn’t changed by the nonce manager. And a similar block with nonce = next_nonce + 13 for when the nonce manager changes it.